src/main/java/com/kolmakova/task2/builder/FlowerBuilderFactory.java

```diff
@@ -0,0 +1,39 @@
+package com.kolmakova.task2.builder;
+
+import com.kolmakova.task2.builder.impl.*;
+import com.kolmakova.task2.exception.ParseXmlException;
+import com.kolmakova.task2.parser.ParserType;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class FlowerBuilderFactory {
+
+    private static final Logger LOGGER = LogManager.getLogger();
+
+    private FlowerBuilderFactory() {
+    }
+
+    public static AbstractFlowersBuilder createFlowerBuilder(String fileName, String schemaName, ParserType parserType) throws ParseXmlException {
```

Comment:
retrun FlowerParser interface.
Return the type, which will be used by clients. And clients will call the methods of FlowerParser